internal/rbd/rbd_util.go

```diff
@@ -951,52 +951,51 @@ func (ri *rbdImage) checkImageChainHasFeature(ctx context.Context, feature uint6
 // the structure with elements from on-disk snapshot metadata as well.
 func genSnapFromSnapID(
 	ctx context.Context,
-	rbdSnap *rbdSnapshot,
 	snapshotID string,
 	cr *util.Credentials,
 	secrets map[string]string,
-) error {
+) (*rbdSnapshot, error) {
 	var vi util.CSIIdentifier
-
-	rbdSnap.VolID = snapshotID
+	rbdSnap := &rbdSnapshot{}
 
```

Comment:
@nixpanic, forgot adding?
```go
rbdSnap.VolID = snapshotID
```

Reply:
how did i miss this one 🤦🏻

Reply:
Oh right, that is done a few lines  further below. Thanks for checking!